C/common/management_client.cpp

```diff
@@ -864,3 +866,210 @@ bool ManagementClient::addAuditEntry(const std::string& code,
 	}
 	return false;
 }
+
+/**
+ * Checks and validate the JWT bearer token
+ *
+ * @param request	HTTP request object
+ * @param claims	Map to fill with JWT public token claims
+ * @return		True on success, false otherwise
+ */
+bool ManagementClient::verifyAccessBearerToken(shared_ptr<HttpServer::Request> request,
+						map<string, string>& claims)
+{
+	string bearer_token = getAccessBearerToken(request);
+
+	if (bearer_token.length() == 0)
+	{
+		return false;
+	}
+
+	bool ret = true;
+
+	// Check token already exists in cache:
+	std::map<std::string, std::string>::iterator item;
+	// Acquire lock
+	m_mtx_rTokens.lock();
+
+	item = m_received_tokens.find(bearer_token);
+	if (item  == m_received_tokens.end())
+	{
+		bool verified = false;
+		// Token does not exist:
+		// Verify it by calling Fledge management endpoint
+		string url = "/fledge/service/verify_token";
+                string payload;
+                SimpleWeb::CaseInsensitiveMultimap header;
+                header.emplace("Authorization", "Bearer " + bearer_token);
+                auto res = this->getHttpClient()->request("POST", url.c_str(), payload, header);
+		Document doc;
+		string response = res->content.string();
+		doc.Parse(response.c_str());
+		if (doc.HasParseError())
+		{
+			bool httpError = (isdigit(response[0]) && isdigit(response[1]) && isdigit(response[2]) && response[3]==':');
+			m_logger->error("%s error in service token verification: %s\n", 
+								httpError?"HTTP error during":"Failed to parse result of", 
+								response.c_str());
+			verified = false;
+		}
+		else
+		{
+			if (doc.HasMember("error"))
+			{
+				string error = doc["error"].GetString();
```

Comment:
Probably should do doc["error'].IsString() as it has a habit of aborting if not.